01-supp-addressingData.Rmd

```diff
@@ -0,0 +1,168 @@
+---
+layout: page
+title: Programming with R
+subtitle: Addressing data
+minutes: 20
+---
+
+
+```{r, include = FALSE}
+source('tools/chunk-options.R')
+```
+
+> ## Objectives {.objectives}
+>
+> Understand the 3 different ways R can address data inside a data frame.
+>
+> Combine different methods for addressing data with the assignment operator to update subsets of data
+
+R is a powerful language for data manipulation. There are 3 main ways for addressing data inside R objects.
+
+* By index (slicing)
+* By logical vector
+* By name (columns and lists only)
+
+Lets start by loading some sample data:
+
+```{r readData}
+dat<-read.csv(file='data/sample.csv',header=TRUE, stringsAsFactors=FALSE)
```

Comment:
I generally err on using defaults. It's easier to do this and perhaps introduce stringsasfactors after running into a gotcha (haven't tested out the lesson, but at first glance it seems everything will run as expected with factors). 

Also, why use `header = TRUE` if this is default?

My impression is that it is much easier